code/ui/inputbox.cpp

```diff
@@ -477,3 +490,33 @@ void UI_INPUTBOX::set_text(const char *in)
 
 	position = in_length;  // fixes the zero-length-I-don't-think-so bug
 }
+
+void UI_INPUTBOX::append_text(const char *in)
+{
+	if (in == nullptr) {
+		return;
+	}
+
+	// current size
+	size_t textlen = strlen(text);
+
+	if (textlen == static_cast<size_t>(length)) {
+		return;
+	}
+
+	// strcat_s() will zero the string if it's too long, which isn't what we want
+	// SDL_strlcat() is safe while still preserving the dest string on error
+	SDL_strlcat(text, in, length+1);
```

Comment:
In the case of overflow, will this paste as much of the text as will fit, or will it just not paste anything?

Reply:
It will paste as much as will fit. I thought about making it an error and leaving the dest string unchanged, but there doesn't seem to be a good way to report that condition to the user.